crates/ty_python_semantic/src/types.rs

```diff
@@ -2995,6 +2995,12 @@ impl<'db> Type<'db> {
                 ))
                 .into()
             }
+            Type::ClassLiteral(class)
+                if name == "__dataclass_fields__" && class.dataclass_params(db).is_some() =>
+            {
+                // Make this class look like a subclass of the `DataClassInstance` protocol
+                Symbol::bound(Type::any()).into()
```

Comment:
Shouldn't this be `dict[str, Field[Any]]`? If we can't spell this out yet, maybe make it a `todo_type!("…")` for now?

Also, I think we should make this a `ClassVar`. Instead of calling `.into()`, you can add the corresponding type qualifier here.

Reply:
Oh, I did not know about `todo_type`! I've updated the type to `dict[str, todo_type("dataclass.Field")]`

Reply:
We should be able to do something like

```rs
KnownClass::Dict.to_specialized_instance(db, [KnownClass::Str.to_instance(db), todo_type("dataclass.Field")])
```

Or, even better:

```rs
KnownClass::Dict.to_specialized_instance(db, [KnownClass::Str.to_instance(db), KnownClass::Field.to_specialized_instance(db, Type::any())])
```

but that requires adding a `KnownClass::Field` variant, which I think is fine to defer to a followup for now if you don't want to do it here

Reply:
Yup, the first one is what I've pushed here.
I can add a new `KnownClass::Field` variant in a follow up PR.

Reply:
Adding `Field` did not seem too difficult, so I added it in this PR itself. Thanks for the suggestion!